services/command-router-base/src/main/java/org/eclipse/hono/commandrouter/impl/CommandRouterServiceImpl.java

```diff
@@ -176,6 +200,46 @@ private Duration getSanitizedLifespan(final Duration lifespan) {
                 .otherwise(t -> CommandRouterResult.from(ServiceInvocationException.extractStatusCode(t)));
     }
 
+    @Override
+    public Future<CommandRouterResult> enableCommandRouting(final List<String> tenantIds, final Span span) {
+        Objects.requireNonNull(tenantIds);
+        final boolean isProcessingRequired = tenantsToEnable.isEmpty();
+        tenantsToEnable.addAll(tenantIds);
+        if (isProcessingRequired) {
+            processTenantQueue();
```

Comment:
I think the tracing output could be structured a bit better by creating a parent span for the "re-enable command routing for tenant" spans created in `processTenantQueue()`.
The span could be created here as a FOLLOWS_FROM span of the `enableCommandRouting` `span` parameter.
(In the `isProcessingRequired` `else` case, the fact that the tenant queue is already being processed could be logged to the `span` (of the `enableCommandRouting` parameter).)

Reply:
Why don't we simply create all of the *re-enable command routing for tenant* spans as *FOLLOWS_FROM* spans of the *enableCommandRouting* span?

Reply:
Would be ok as well.